lib/excon.rb

```diff
@@ -109,6 +109,7 @@ def new(url, params = {})
       end
       params = {
         :host       => uri.host,
+        :hostname   => uri.hostname,
```

Comment:
All of the calls to URI#hostname will fail on ruby 1.9.2 and earlier because it wasn't added until 1.9.3.  I can add conditional logic around these calls if we want to support 1.9.2 and earlier.